### PR TITLE
fix endbunny and endgame UMAPINFO fields

### LINKED ISSUE #286: 'endbunny' and 'endgame' fields don't work properly (Doom 1, UMAPINFO)
Test WAD: [endbunnyD1.zip](https://github.com/coelckers/prboom-plus/files/6353146/endbunnyD1.zip)
Exit on E1M2 should end the game and show the bunny scroller. Exit on E1M3 should end the episode.

### PR DESCRIPTION
**Pull request by rfomin**
Fixes #286

### CODE REVIEW

File: prboom2/src/umapinfo.cpp [hunk 1]
```diff
@@ -630,7 +630,6 @@ int ParseUMapInfo(const unsigned char *buffer, size_t length, umapinfo_errorfunc
 		if (parsed.endpic[0] && (strcmp(parsed.endpic, "-") != 0))
 		{
 			parsed.nextmap[0] = parsed.nextsecret[0] = 0;
-			if (parsed.endpic[0] == '!') parsed.endpic[0] = 0;
```
Comment (JadingTsunami) [inline prboom2/src/umapinfo.cpp:633]: Isn't the purpose of this line to handle `endgame = true`? If it is removed, how does that option work?

**Comment by JadingTsunami** [comment]
Looks good to me.

File: prboom2/src/umapinfo.cpp (hunk 1 above)
Comment (fabiangreffrath) [inline prboom2/src/umapinfo.cpp:633]: Why was this sucker line there in the first place?

Comment (rfomin) [reply prboom2/src/umapinfo.cpp:633]: Eh, I don't know, just a bug.

File: prboom2/src/g_game.c [hunk 2]
```diff
@@ -1630,7 +1630,7 @@ void G_DoCompleted (void)
   if (gamemapinfo)
   {
 	  const char *next = "";
-	  if (gamemapinfo->endpic[0] && (strcmp(gamemapinfo->endpic, "-") != 0) && gamemapinfo->nointermission)
+	  if (gamemapinfo->endpic[0] && (strcmp(gamemapinfo->endpic, "-") != 0) && !gamemapinfo->nointermission)
```
Comment (fabiangreffrath) [inline prboom2/src/g_game.c:1633]: Wait a minute, I just skimmed through this code yesterday. Why didn't I catch this?

This isn't directly related to the original problem in #286, right?

To me this reads like "if we have set an endpic and nointermission is not set, we should skip the intermission screen" which just doesn't sound right.

Comment (rfomin) [reply prboom2/src/g_game.c]: Yes, you are right, I missed one important part of the Woof UMPAINFO implementation and got confused.

**Comment by fabiangreffrath** [comment]
Thank you!

**Pull request merged**